doc/syntax.rst

```diff
@@ -288,3 +288,37 @@ as an array:
   they appear in the blackbird program
 
 * each row represents a shot/run
+
+Templates
+---------
+
+A Blackbird template is simply a Blackbird script that contains **template parameters**.
+
+Template parameters use the syntax ``{parameter_name}``, and can be placed within any arithmetic expression.
```

Comment:
```suggestion
Template parameters use the syntax ``{parameter_name}``, and can be placed within any numeric expression.
```

Reply:
So things like `Dgate({a} * 4)` will work?

Reply:
Yep! Any expression will work. For instance, `Dgate({a}*4+sin({b}))` will also work.